Add a test to verify that only one instance of stratisd can be running

mulkieran opened the pull request:
Closes #1644.

mulkieran commented:
Had to allow an "unnecessary lambda"

jbaublitz commented on tests/client-dbus/tests/misc/test_startup.py:
```diff
@@ -0,0 +1,63 @@
+        )
+
+        # pylint: disable=unnecessary-lambda
+        self.addCleanup(lambda: self.first_process.terminate())
```
Is the reason that this is:
```self.addCleanup(lambda: self.first_process.terminate())```
and not:
```self.addCleanup(self.first_process.terminate)```
just for readability purposes?

mulkieran replied:
No, I think it's a hold over from a previous edit. I should be able to get rid of it. Let me give that a try.

mulkieran commented:
rebased, and squashed removal of the pylint directive, see comment: https://github.com/stratis-storage/stratisd/pull/1643#pullrequestreview-294430733.
